File: 167-Two-Sum-II-Input-Array-Is-Sorted.py
class Solution(object):
	def twoSum(self, numbers, target):
		l = 0
		r = len(numbers) - 1

		while l < r:
			if numbers[r] + numbers[l] > target:
				r -= 1
			elif numbers[r] + numbers[l] < target:
				l += 1
			else:
				return [l + 1, r + 1]

File: neetcode/167-Two-Sum-II---Input-Array-Is-Sorted.ts
function twoSum(numbers: number[], target: number): number[] {
  let l = 0, r = numbers.length - 1

  while (l < r) {
    const sum = numbers[r] + numbers[l]

    if (sum > target) r--
    else if (sum < target) l++
    else return [l + 1, r + 1]
  }

  return [-1]
};